modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/OrchestratorService.java

```diff
@@ -78,19 +84,25 @@ public void registerManagedService(String oclLocation) throws Exception {
      *
      * @param ocl the OCL descriptor.
      */
+    @Async("taskExecutor")
     public void registerManagedService(Ocl ocl) {
+        if (this.databaseOrchestratorStorage.isExists(ocl.getName())) {
+            throw new RuntimeException(
+                "Managed service " + ocl.getName() + " already registered.");
+        }
         if (plugins.isEmpty()) {
             log.warn("No plugins available. Request ignored.");
-            return;
+            throw new RuntimeException("No plugins available.");
         }
-
+        Thread.currentThread().setName(Thread.currentThread().getName() +
```

Comment:
Do we really need to add the thread names to be updated? 
I prefer that we dont touch the thread name and instead add a MDC  for plugin name in the log line.

Reply:
I add plugin name and service name into the thread name now.